consensus/hotstuff/cruisectl/transition_time_test.go

```diff
@@ -0,0 +1,75 @@
+package cruisectl
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestParseTransition_Valid tests that valid transition configurations have
+// consistent parsing and formatting behaviour.
+func TestParseTransition_Valid(t *testing.T) {
+	cases := []struct {
+		transition EpochTransitionTime
+		str        string
+	}{{
+		transition: EpochTransitionTime{time.Sunday, 0, 0},
+		str:        "sunday@00:00",
+	}, {
+		transition: EpochTransitionTime{time.Wednesday, 8, 1},
+		str:        "wednesday@08:01",
+	}, {
+		transition: EpochTransitionTime{time.Monday, 23, 59},
+		str:        "monday@23:59",
+	}}
+
+	for _, c := range cases {
+		t.Run(c.str, func(t *testing.T) {
+			// 1 - the computed string representation should match the string fixture
+			assert.Equal(t, c.str, c.transition.String())
+			// 2 - the parsed transition should match the transition fixture
+			parsed, err := ParseTransition(c.str)
+			assert.NoError(t, err)
+			assert.Equal(t, c.transition, *parsed)
+		})
+	}
+}
+
+// TestParseTransition_Invalid tests that a selection of invalid transition strings
+// fail validation and return an error.
+func TestParseTransition_Invalid(t *testing.T) {
```

Comment:
It's a great application of rapid framework to generate possible valid and invalid variations of input strings.

Reply:
Yes! I should have thought of that.